### PR TITLE
CK3 Provinces with tests

### PR DESCRIPTION
**Pull request by IhateTrains**

### CODE REVIEW

File: ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	public class ProvinceDetails {
+		// These values are open to ease management.
+		// This is a storage container for CK3::Province.
+		public string Culture { get; set; } = string.Empty;
+		public string Religion { get; set; } = string.Empty;
+		public string Holding { get; set; } = "none";
+		public List<string> Buildings { get; } = new();
+
+		public ProvinceDetails() { }
+		public ProvinceDetails(ProvinceDetails otherDetails) {
```
Comment (Idhrendur) [inline ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs:15]: Test this

File: ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+			if (cultureOpt is not null) {
+				Culture = cultureOpt;
+			}
+			var religionOpt = history.GetSimpleFieldValue("religion", date);
+			if (religionOpt is not null) {
+				Religion = religionOpt;
+			}
+			var holdingOpt = history.GetSimpleFieldValue("holding", date);
+			if (holdingOpt is null) {
+				Logger.Warn("Province's holding can't be null!");
```
Comment (Idhrendur) [inline ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs:35]: Consider testing this

File: ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+			} else {
+				Holding = holdingOpt;
+			}
+			var buildingsOpt = history.GetContainerFieldValue("buildings", date);
+			if (buildingsOpt is null) {
+				Logger.Warn("Province's buildings list can't be null!");
```
Comment (Idhrendur) [inline ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs:41]: And this

File: ImperatorToCK3/CK3/Provinces/ProvinceMappings.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	/// 	religion = coptic
+	/// 	culture = coptic
+	/// }
+	/// 
+	/// Example province_mapping in game/history/province_mapping:
+	/// 6874 = 6872
+	/// 
+	/// Now 6874 history is same as 6872 history.
+	/// </summary>
+	class ProvinceMappings : Parser {
```
Comment (Idhrendur) [inline ImperatorToCK3/CK3/Provinces/ProvinceMappings.cs:20]: Tests

**Review by Idhrendur — APPROVED** [review]

File: ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs (hunk 2 above)
Comment (IhateTrains) [reply ImperatorToCK3/CK3/Provinces/ProvinceDetails.cs:35]: I can't really test it, as the initial value for holding in historyFactory is "none", so the returned value will never be null. I added the null check to get rid of the IDE warning.

**Pull request merged**